File: python/can-place-flowers.py
class Solution:
    def is_no_flower(i: int, flowerbed: List[int]):
        return i < 0 or i == len(flowerbed) or flowerbed[i] == 0

    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True
        for i in range(len(flowerbed)):
            if flowerbed[i] == 0 and Solution.is_no_flower(i - 1, flowerbed) and Solution.is_no_flower(i + 1, flowerbed):
                flowerbed[i] = 1
                n -= 1
                if n == 0:
                    return True
        return False

File: rust/can-place-flowers.rs
impl Solution {
    pub fn can_place_flowers(mut flowerbed: Vec<i32>, mut n: i32) -> bool {
        if n == 0 {
            return true;
        }
        let len = flowerbed.len();
        let mut i = 0;
        while i < len {
            if flowerbed[i] == 0 && (i == 0 || flowerbed[i - 1] == 0) && (i == flowerbed.len() - 1 || flowerbed[i + 1] == 0) {
                if let Some(elem) = flowerbed.get_mut(i) {
                    *elem = 1;
                    n -= 1;
                    if n == 0 {
                        return true;
                    }
                }
            }
            i += 1;
        }
        return false;
    }
}
